### PR TITLE
Bucket Sort using C++

### LINKED ISSUE #124: Add Sorting Algorithms
- [x] C++
    - [ ] Bubble Sort
    - [ ] Insertion Sort
    - [ ] Selection Sort
    - [ ] Merge Sort
    - [ ] Quick Sort
 - [x] Java
    - [x] Bubble Sort
    - [x] Insertion Sort
    - [x] Selection Sort
    - [ ] Merge Sort
    - [ ] Quick Sort
- [x] Python
    - [ ] Bubble Sort
    - [ ] Insertion Sort
    - [ ] Selection Sort
    - [ ] Merge Sort
    - [ ] Quick Sort

### PR DESCRIPTION
**Pull request by EkjotKaur**
Code for Bucket Sort in C++
with commented

- definition
- input and output
- time & space complexity

bucketSort

Fixed #124
Please view my PR @tarun26091999

### CODE REVIEW

File: Code/C++/bucket_sort.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    cin >> A[i];
+  }
+
+  BucketSort(A, n);
+
+  for (int i = 0; i < n; i++)
+  {
+    cout << A[i] << " ";
+  }
+
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:61]: remove extra line

File: Code/C++/bucket_sort.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  return 0;
+}
+
+/*
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:65]: Correct the indentation of the code and properly align it too.

**Review by raksha009 — CHANGES_REQUESTED** [review]
Read the guidelines as well and update the required readme (PS: not the main one).

**Comment by EkjotKaur** [comment]
@tarun26091999 @nakul-19 @raksha009 please see this PR

File: Code/C++/bucket_sort.cpp [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+    Test Case:
+    Input:    7
+              4 63 1 53 87 44 36
+    Output:   1 4 36 44 53 63 87
+
+    Time Complexity:
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:65]: Specify what is n and k

File: Code/C++/bucket_sort.cpp [hunk 4]
```diff
@@ -0,0 +1,72 @@
+
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:1]: Remove this extra line

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by EkjotKaur** [comment]
@tarun26091999 @nakul-19 @raksha009 please see this PR

**Comment by raksha009** [comment]
Use 4 space(a tab) indentation throughout,

**Review by raksha009 — CHANGES_REQUESTED** [review]
Plz use 4 space indentation everywhere

**Comment by EkjotKaur** [comment]
I have used 4 space indentation everywhere.

File: Code/C++/bucket_sort.cpp [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  Bucket Sort is a sorting technique that sorts the elements by first dividing the elements into several groups called buckets. 
+  The elements inside each bucket are sorted using any of the suitable sorting algorithms or recursively calling the same algorithm.
+*/
+
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+int maxValue(int A[], int n)
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:12]: Add necessary comments everywhere - like what does this function do etc.

File: Code/C++/bucket_sort.cpp [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	{
+		for (int j = Bins[i].size() - 1; j >= 0; j--)
+			A[k++] = Bins[i][j];
+		i++;
+	}
+}
+
+int main()
+{
+	int n;
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:47]: Add comments.

File: Code/C++/bucket_sort.cpp [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+	for (int i = 0; i < n; i++)
+		cout << A[i] << " ";
+	return 0;
+}
+
+/*
+
+    Test Case:
+    Input:   	7
+              	4 63 1 53 87 44 36
```
Comment (raksha009) [inline Code/C++/bucket_sort.cpp:62]: Add 2 test cases

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by anishsingh935** [comment]
HI can I implement the code of this problem with understandable comments??

**Comment by EkjotKaur** [comment]
I was added all the suitable comments and 2 test cases
@raksha009 please merge this PR

**Review by raksha009 — APPROVED** [review]
LGTM 👍🏼

**Comment by EkjotKaur** [comment]
@rudrakshi99 Please review my PR.

**Comment by EkjotKaur** [comment]
@rudrakshi99 Please review my PR.

**Comment by einhawking** [comment]
I am a participant of GSSOC'21
I would like to contribute in this

**Comment by EkjotKaur** [comment]
@OjusWiZard  Please view this PR.

**Pull request merged**